ss/pebbledb/iterator.go

```diff
@@ -154,6 +154,27 @@ func (itr *iterator) Next() {
 		// append the current iterator key to the prefix and seek to that key.
 		itr.valid = itr.source.SeekLT(MVCCEncode(nextKey, itr.version+1))
 
+		tmpKey, _, ok := SplitMVCCKey(itr.source.Key())
+		if !ok {
+			// XXX: This should not happen as that would indicate we have a malformed
+			// MVCC key.
+			itr.valid = false
+			return
+		}
+
+		// There exists cases where the SeekLT() call moved us back to the same key
+		// we started at, so we must move to next key, i.e. two keys forward.
+		if bytes.Equal(tmpKey, currKey) {
+			if itr.source.SeekGE(MVCCEncode(nextKey, 0)) {
+				if itr.source.NextPrefix() {
+					itr.Next()
```

Comment:
Why do we move to next again after seeking GE, will that result in skipping a key that we should iterate?